spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -348,17 +368,55 @@ function onRecvPacket(packet: Packet) {
     } else {
       // since we are receiving to a new sink zone we prepend the prefix to the trace
       prefix = "{packet.destPort}/{packet.destChannel}/"
-      newTrace = append([]string{prefix}, token.trace...)
-      onChainDenom = constructOnChainDenom(newTrace, token.denom)
+      onChainTrace = append([]string{prefix}, token.trace...)
+      onChainDenom = constructOnChainDenom(onChainTrace, token.denom)
       // sender was source, mint vouchers to receiver (assumed to fail if balance insufficient)
-      err = bank.MintCoins(data.receiver, onChainDenom, token.amount)
+      err = bank.MintCoins(receiver, onChainDenom, token.amount)
       if (err !== nil) {
         ack = FungibleTokenPacketAcknowledgement{false, "mint coins failed"}
         // break out of for loop on first error
         break
       }
     }
+
+    // add the received token to the received tokens list
+    recvToken = Token{
+      denom: token.denom,
+      trace: onChainTrace,
+      amount: token.amount,
+    }
+    receivedTokens = append(receivedTokens, recvToken)
+  }
+
+  // if there is an error ack return immediately and do not forward further
+  if !ack.Success() {
+    return ack
   }
+
+  // if acknowledgement is successful and forwarding path set
+  // then start forwarding
+  if forwardingPath != nil {
+    nextPort, nextChannel = split(forwardingPath[0], "/")
+    // send the tokens we received above to the next port and channel
+    // on the forwarding path
+    // and reduce the forwardingPath by the first element
+    nextSequence = sendFungibleTokens(
```

Comment:
in this call to `sendFungibleTokens`, when the chain is source of the tokens, we do a bank transfer where sender and receiver are the same address (i.e. the escrow account on the source channel). Just mentioning, in case there's a way to avoid the extra bank transfer.